packages/react-components/theme-designer/src/components/Sidebar/Form.tsx

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable react/jsx-no-bind */
+import * as React from 'react';
+import { makeStyles, shorthands } from '@griffel/react';
+import { Label, Input, Slider, tokens } from '@fluentui/react-components';
+import { useDebounce } from '../../utils/useDebounce';
+
+import type { CustomAttributes, DispatchTheme } from '../../useThemeDesignerReducer';
+
+const useStyles = makeStyles({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    ...shorthands.gap(tokens.spacingVerticalXXL, tokens.spacingHorizontalXXL),
+  },
+  inputs: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'flex-start',
+  },
+  keyColor: {
+    paddingLeft: '0px',
+  },
+  labels: {
+    display: 'grid',
+    gridTemplateColumns: '135px 30px',
+    columnGap: '15px',
+  },
+  colorPicker: {
+    ...shorthands.border('1px', 'solid', tokens.colorNeutralStroke1),
+    ...shorthands.borderRadius('25px'),
+    height: '30px',
+    width: '30px',
+    ...shorthands.overflow('hidden'),
+  },
+  color: {
+    ...shorthands.padding('0px'),
+    ...shorthands.border('0px'),
+    opacity: '0',
+  },
+  slider: {
+    display: 'grid',
+    gridTemplateColumns: '115px 50px',
+    columnGap: '15px',
+  },
+});
+
+export interface FormProps {
+  sidebarId: string;
+  dispatchAppState: React.Dispatch<DispatchTheme>;
+  formState: CustomAttributes;
+  setFormState: React.Dispatch<CustomAttributes>;
+}
+
+export const Form: React.FC<FormProps> = props => {
+  const styles = useStyles();
+
+  const { sidebarId, dispatchAppState, formState, setFormState } = props;
+
+  const formReducer = (state: CustomAttributes, action: { attributes: CustomAttributes }) => {
```

Comment:
this function can be moved out of the Form component, up to line 53. it only needs to be defined once.